### PR TITLE
CTCP-1619 Add Another Border Transport Page

### PR DESCRIPTION
**Pull request by Reece-Carruthers**

### CODE REVIEW

File: app/pages/sections/TransportSection.scala [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package pages.sections
+
+import play.api.libs.json.JsPath
+import pages.QuestionPage
+
+case object TransportSection extends QuestionPage[Nothing] {
```
Comment (jackheslop96) [inline app/pages/sections/TransportSection.scala:22]: Delete

File: app/pages/transport/transportMeans/active/AddAnotherBorderTransportPage.scala [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+import controllers.transport.transportMeans.active.routes
+import models.{Mode, UserAnswers}
+import pages.QuestionPage
+import pages.sections.TransportSection
+import play.api.libs.json.JsPath
+import play.api.mvc.Call
+
+case object AddAnotherBorderTransportPage extends QuestionPage[Boolean] {
+
+  override def path: JsPath = TransportSection.path \ toString
```
Comment (jackheslop96) [inline app/pages/transport/transportMeans/active/AddAnotherBorderTransportPage.scala:28]: I think this can be deleted? We don't save their answer to the 'add another' question

File: app/views/transport/transportMeans/active/AddAnotherBorderTransportView.scala.html [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+    )
+
+    @listWithActions(viewModel.listItems)
+
+    @formHelper(action = viewModel.onSubmitCall, Symbol("autoComplete") -> "off") {
+
+        @if(viewModel.allowMoreActiveBorderTransports) {
+            @inputYesNo(
+                field = form("value"),
+                yesNoType = YesNoWithLegendAndHint(
```
Comment (jackheslop96) [inline app/views/transport/transportMeans/active/AddAnotherBorderTransportView.scala.html:57]: `YesNoWithLegendAndHint` can be deleted. inputYesNo takes a hint as one of the parameters

File: app/views/components/InputYesNo.scala.html [hunk 4]
```diff
@@ -63,4 +64,9 @@
     case YesNoWithLegend(legend) => {
         @govukRadios(radios.withLegend(legend))
     }
+
+    case YesNoWithLegendAndHint(hintText, legend) => {
```
Comment (jackheslop96) [inline app/views/components/InputYesNo.scala.html:68]: Delete (see comment in AddAnotherBorderTransportView)

File: app/viewModels/components/InputYesNoViewModel.scala [hunk 5]
```diff
@@ -37,4 +37,9 @@ object InputYesNoViewModel {
   case class YesNoWithLegend(
     legend: String
   ) extends InputYesNoViewModel
+
+  case class YesNoWithLegendAndHint(
```
Comment (jackheslop96) [inline app/viewModels/components/InputYesNoViewModel.scala:41]: Delete (see comment in AddAnotherBorderTransportView)

File: app/views/utils/ViewUtils.scala [hunk 6]
```diff
@@ -58,6 +58,20 @@ object ViewUtils {
         fieldset = Some(Fieldset(legend = Some(Legend(content = Text(legend), classes = legendClass, isPageHeading = false))))
       )
     }
+
+    def withLegendAndHint(legend: String, legendIsVisible: Boolean = true, hintText: String): Radios = {
```
Comment (jackheslop96) [inline app/views/utils/ViewUtils.scala:62]: Delete (see comment in AddAnotherBorderTransportView)

File: conf/application.conf [hunk 7]
```diff
@@ -186,6 +186,7 @@ limits {
   maxCountriesOfRouting = 99
   maxOfficesOfTransit = 9
   maxOfficesOfExit = 9
+  maxActiveBorderTransports = 9
```
Comment (jackheslop96) [inline conf/application.conf:189]: Can we also set this in test.application.conf? Not so bad when it's 9 but for larger numbers it can be quite exhaustive in the tests

File: test/base/SpecBase.scala [hunk 8]
```diff
@@ -67,7 +67,7 @@ trait SpecBase
   def messagesApi: MessagesApi    = injector.instanceOf[MessagesApi]
   implicit def messages: Messages = messagesApi.preferred(fakeRequest)
 
-  def frontendAppConfig: FrontendAppConfig = injector.instanceOf[FrontendAppConfig]
+  implicit def frontendAppConfig: FrontendAppConfig = injector.instanceOf[FrontendAppConfig]
```
Comment (jackheslop96) [inline test/base/SpecBase.scala:70]: Is this change needed? Can cause issues if there is another implicit config defined in scope

**Review by kevincoatsworth — APPROVED** [review]
